### ISSUE TITLE
Jekyll build errors on M1 mac

### ISSUE THREAD

**Issue by holdenmatt** [duckdb/duckdb-web#1090]
I spent a few hours in dependency hell trying to install ruby 2.7.3 and jekyll on an M1 Pro mac. I kept getting this error:
```
Unable to load the EventMachine C extension; To use the pure-ruby reactor, require 'em/pure_ruby'
bundler: failed to load command: jekyll (/Users/matt/.gem/ruby/2.7.3/bin/jekyll)
Traceback (most recent call last):
	35: from /Users/matt/.rubies/ruby-2.7.3/bin/bundle:23:in `<main>'
	34: from /Users/matt/.rubies/ruby-2.7.3/bin/bundle:23:in `load'
	33: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/exe/bundle:29:in `<top (required)>'
	32: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	31: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/exe/bundle:37:in `block in <top (required)>'
	30: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/cli.rb:28:in `start'
	29: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	28: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/cli.rb:34:in `dispatch'
	27: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	26: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	25: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	24: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/cli.rb:492:in `exec'
	23: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/cli/exec.rb:23:in `run'
	22: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/cli/exec.rb:58:in `kernel_load'
	21: from /Users/matt/.gem/ruby/2.7.3/gems/bundler-2.4.19/lib/bundler/cli/exec.rb:58:in `load'
	20: from /Users/matt/.gem/ruby/2.7.3/bin/jekyll:23:in `<top (required)>'
	19: from /Users/matt/.gem/ruby/2.7.3/bin/jekyll:23:in `load'
	18: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/exe/jekyll:15:in `<top (required)>'
	17: from /Users/matt/.gem/ruby/2.7.3/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	16: from /Users/matt/.gem/ruby/2.7.3/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	15: from /Users/matt/.gem/ruby/2.7.3/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	14: from /Users/matt/.gem/ruby/2.7.3/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	13: from /Users/matt/.gem/ruby/2.7.3/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	12: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:75:in `block (2 levels) in init_with_program'
	11: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:93:in `start'
	10: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:93:in `each'
	 9: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:93:in `block in start'
	 8: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:101:in `process'
	 7: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:147:in `register_reload_hooks'
	 6: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve.rb:147:in `require_relative'
	 5: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve/live_reload_reactor.rb:3:in `<top (required)>'
	 4: from /Users/matt/.gem/ruby/2.7.3/gems/jekyll-3.9.3/lib/jekyll/commands/serve/live_reload_reactor.rb:3:in `require'
	 3: from /Users/matt/.gem/ruby/2.7.3/gems/em-websocket-0.5.3/lib/em-websocket.rb:3:in `<top (required)>'
	 2: from /Users/matt/.gem/ruby/2.7.3/gems/em-websocket-0.5.3/lib/em-websocket.rb:3:in `require'
	 1: from /Users/matt/.gem/ruby/2.7.3/gems/eventmachine-1.2.7/lib/eventmachine.rb:8:in `<top (required)>'
/Users/matt/.gem/ruby/2.7.3/gems/eventmachine-1.2.7/lib/eventmachine.rb:8:in `require': dlopen(/Users/matt/.gem/ruby/2.7.3/gems/eventmachine-1.2.7/lib/rubyeventmachine.bundle, 0x0009): symbol not found in flat namespace '_SSL_get1_peer_certificate' - /Users/matt/.gem/ruby/2.7.3/gems/eventmachine-1.2.7/lib/rubyeventmachine.bundle (LoadError)
```

In case it helps someone else, [this comment](https://github.com/postmodern/ruby-install/issues/409#issuecomment-1681133425) is what finally fixed it for me.

tl;dr:
1. `export PKG_CONFIG_PATH="$(brew --prefix openssl@1.1)/lib/pkgconfig"`
2. `ruby-install 2.7.3 -- --with-openssl-dir=$(brew --prefix openssl@1.1)`
3. `gem uninstall eventmachine`
4. `gem install eventmachine`
5. `bundle exec jekyll serve`

**Comment by szarnyasg** [comment]
@holdenmatt many thanks for providing these instructions! A few quick comments:

1) I also have an Apple Silicon Mac. I rely on [RVM](https://rvm.io/) for installing Ruby and I ended up using a very similar workaround: `rvm install 2.7.3 --with-openssl-dir=/opt/homebrew/opt/openssl@1.1`

2) I considered adding this workaround to the build instructions but I was on the fence (because it's only affecting M1 users). I think it's worth adding.

3) In general, setting up Ruby and Jekyll can be challenging, so we have a Docker container to make the process easier:
https://github.com/duckdb/duckdb-web/blob/main/building.md#serve-the-site-from-docker

**Comment by holdenmatt** [comment]
Thanks! If there had been a note to that effect, it would have saved me a few hours of frustration :)

Maybe just add a brief note at the top of this doc (similar to the note about building on Windows)?
https://github.com/duckdb/duckdb-web/blob/main/building.md
